AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_HairAdded.py

```diff
@@ -0,0 +1,182 @@
+
+    from editor_python_test_tools.asset_utils import Asset
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper
+    from Atom.atom_utils.atom_constants import AtomComponentProperties
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        TestHelper.init_idle()
+        TestHelper.open_level("Graphics", "base_empty")
+
+        # Test steps begin.
+        # 1. Create a Hair entity with no components.
+        hair_entity = EditorEntity.create_editor_entity(AtomComponentProperties.hair())
+        Report.critical_result(Tests.hair_creation, hair_entity.exists())
+
+        # 2. Add a Hair component to Hair entity.
+        hair_component = hair_entity.add_component(AtomComponentProperties.hair())
+        Report.critical_result(
+            Tests.hair_component,
+            hair_entity.has_component(AtomComponentProperties.hair()))
+
+        # 3. Remove Hair component.
+        hair_component.remove()
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.hair_component_removal,
+            not hair_entity.has_component(AtomComponentProperties.hair()))
+
+        # 4. UNDO the component removal.
+        general.undo()
+        general.idle_wait_frames(1)
+        Report.critical_result(
+            Tests.hair_component,
+            hair_entity.has_component(AtomComponentProperties.hair()))
+
+        # 5. Verify Hair component not enabled.
+        Report.result(Tests.hair_disabled, not hair_component.is_enabled())
+
+        # 6. Add Actor component since it is required by the Hair component.
+        actor_component = hair_entity.add_component(AtomComponentProperties.actor())
+        Report.result(Tests.actor_component, hair_entity.has_component(AtomComponentProperties.actor()))
+
+        # 7. Verify Hair component is enabled.
+        Report.result(Tests.hair_enabled, hair_component.is_enabled())
+
+        # 8. Set assets for Actor and Hair
+        head_path = os.path.join('testdata', 'headbonechainhairstyle', 'test_hair_bone_chain_head_style.azmodel')
+        head_asset = Asset.find_asset_by_path(head_path)
+        hair_path = os.path.join('testdata', 'headbonechainhairstyle', 'test_hair_bone_chain_head_style.tfxhair')
+        hair_asset = Asset.find_asset_by_path(hair_path)
+        actor_component.set_component_property_value(AtomComponentProperties.actor('Actor asset'), head_asset.id)
+        hair_component.set_component_property_value(AtomComponentProperties.hair('Hair Asset'), hair_asset.id)
+        Report.result(
+            Tests.hair_asset,
+            hair_component.get_component_property_value(AtomComponentProperties.hair('Hair Asset')) == hair_asset.id)
+
+        # 23. Enter/Exit game mode.
+        TestHelper.enter_game_mode(Tests.enter_game_mode)
```

Comment:
Numbering is off from this point down, jumps from 8 to 23

Reply:
fixed